src/shogun/kernel/CombinedKernel.cpp

```diff
@@ -144,13 +126,33 @@ bool CCombinedKernel::init(CFeatures* l, CFeatures* r)
 		}
 		else
 		{
-			SG_DEBUG("Initializing 0x%p - \"%s\" (skipping init, this is a CUSTOM kernel)\n", this, k->get_name())
+			SG_DEBUG(
+			    "Initializing 0x%p - \"%s\" (skipping init, this is a CUSTOM "
+			    "kernel)\n",
+			    this, k->get_name())
 			if (!k->has_features())
-				SG_ERROR("No kernel matrix was assigned to this Custom kernel\n")
+				SG_ERROR(
+				    "No kernel matrix was assigned to this Custom kernel\n")
+
+			// clear all previous subsets
+			((CCustomKernel*)k)->remove_all_row_subsets();
```

Comment:
instead of the explicit cast could we do a dynamic_cast here so that way we can clearly get a bad_cast exception in case of some weird `k`?